cdap-distributions/src/etc/cdap/conf.dist/cdap-site.xml

```diff
@@ -20,5 +20,21 @@
     Your site level configuration goes here 
   -->
 
+  <property>
+    <name>dataset.service.exec.threads</name>
+    <value>300</value>
```

Comment:
This was the suggestion for CDAP 3.3. Many performance changes have gone into Dataset framework after that.